core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSupervisor.scala

```diff
@@ -91,8 +96,13 @@ class SubmissionSupervisor(executionServiceCluster: ExecutionServiceCluster,
       case _ => Restart
     }
 
-    def thresholdFunc(cause: Throwable, count: Int): Unit = {
-      logger.error(s"error monitoring submission after $count times", cause)
+    def thresholdFunc(throwable: Throwable, count: Int): Unit = throwable match {
+      case MonitoredSubmissionException(workspaceName, submissionId, cause) =>
+        // increment smaRestart counter
+        restartCounter(workspaceName, submissionId, cause) += 1
+        logger.error(s"error monitoring submission $submissionId in workspace $workspaceName after $count times", cause)
+      case _ =>
+        logger.error(s"error monitoring submission after $count times", throwable)
```

Comment:
Not sure passing an `ActorRef` to `thresholdFunc` would work because you can't peek inside an ActorRef. You could send a message to it, but not sure that would work if the actor had just died.